app/server/api/user_api.py

```diff
@@ -131,12 +131,12 @@ def post(self, user_id):
         )
 
         if response_code == 200:
-            db.session.commit()
-
             user = response_object['data']['user']
             if organisation is not None and user is not None:
                 organisation.send_welcome_sms(user)
 
+            db.session.commit()
```

Comment:
if the commit happens not last, then if `organisation.send_welcome_sms(user)` fails, the api will return a `500` or whatever, so the front end will say the user isn't created when they actually have been.